packages/react/src/components/Tag/index.tsx

```diff
@@ -18,14 +45,79 @@ TagLabel.propTypes = {
   className: PropTypes.string
 };
 
-const Tag = ({ children, className, ...other }: TagProps) => (
-  <div className={classNames('Tag', className)} {...other}>
-    {children}
-  </div>
-);
+const Tag = (props: TagProps) => {
+  // render dismiss Tag
+  if (props.variant === 'dismiss') {
+    const [show, setShow] = useState(true);
```

Comment:
It seems like it would be good if `show` came in via a prop, similar to how the toggle variant gets `toggleState`. It would be more consistent and flexible.